packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -631,7 +631,24 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
   @override
   void didUpdateWidget(covariant DraggableScrollableSheet oldWidget) {
     super.didUpdateWidget(oldWidget);
-    _replaceExtent();
+    if (_replacedExtent()
+      && widget.snap
+      && (widget.snap != oldWidget.snap || widget.snapSizes != oldWidget.snapSizes)
+      && _scrollController.hasClients
+      ) {
+      // Trigger a snap in case snap or snapSizes has changed and there is a
+      // scroll position currently attached. We put this in a post frame
+      // callback so that `build` can update `_extent.availablePixels` before
+      // this runs-we can't use the previous extent's available pixels as it may
+      // have changed when the widget was updated.
+      WidgetsBinding.instance.addPostFrameCallback((Duration timeStamp) {
+        for (int index = 0; index < _scrollController.positions.length; index++) {
+          final _DraggableScrollableSheetScrollPosition position =
+            _scrollController.positions.elementAt(index) as _DraggableScrollableSheetScrollPosition;
+          position.goBallistic(0);
+        }
+      });
+    }
```

Comment:
nit: _replacedExtent is only used here. I think it'd make sense to either inline the function here or to keep all this logic in the function.